src/display/annotation_layer.js

```diff
@@ -530,6 +539,107 @@ var TextWidgetAnnotationElement = (
 
 /**
  * @class
+ * @alias CheckboxWidgetAnnotationElement
+ */
+var CheckboxWidgetAnnotationElement =
+  (function CheckboxWidgetAnnotationElementClosure() {
+
+  function CheckboxWidgetAnnotationElement(parameters) {
+    WidgetAnnotationElement.call(this, parameters);
+  }
+
+  Util.inherit(CheckboxWidgetAnnotationElement, WidgetAnnotationElement, {
+    /**
+     * Render the text widget annotation's HTML element in the empty container.
+     *
+     * @public
+     * @memberof CheckboxWidgetAnnotationElement
+     * @returns {HTMLSectionElement}
+     */
+    render: function CheckboxWidgetAnnotationElement_render() {
+      this.container.className = 'checkboxWidgetAnnotation';
+
+      if (this.renderInteractiveForms) {
+        var element = document.createElement('div');
+        element.addEventListener('click', this._toggle.bind(this, element));
+        this.container.appendChild(element);
+      }
+
+      return this.container;
+    },
+
+    /**
+     * Toggle the checkbox.
+     *
+     * @private
+     * @memberof CheckboxWidgetAnnotationElement
+     */
+    _toggle: function CheckboxWidgetAnnotationElement_toggle(element) {
+      element.className = element.className ? '' : 'checked';
+    }
+  });
+
+  return CheckboxWidgetAnnotationElement;
+})();
+
+/**
+ * @class
+ * @alias RadioButtonWidgetAnnotationElement
+ */
+var RadioButtonWidgetAnnotationElement =
+  (function RadioButtonWidgetAnnotationElementClosure() {
+
+  function RadioButtonWidgetAnnotationElement(parameters) {
+    WidgetAnnotationElement.call(this, parameters);
+  }
+
+  Util.inherit(RadioButtonWidgetAnnotationElement, WidgetAnnotationElement, {
+    /**
+     * Render the text widget annotation's HTML element in the empty container.
+     *
+     * @public
+     * @memberof RadioButtonWidgetAnnotationElement
+     * @returns {HTMLSectionElement}
+     */
+    render: function RadioButtonWidgetAnnotationElement_render() {
+      this.container.className = 'radioButtonWidgetAnnotation';
+
+      if (this.renderInteractiveForms) {
+        var element = document.createElement('div');
+        element.setAttribute('data-parent-name',
+          this.data.fullName.split('`')[0]);
+        element.addEventListener('click', this._toggle.bind(this, element));
+        this.container.appendChild(element);
+      }
+
+      return this.container;
+    },
+
+    /**
+     * Toggle the radio button and unselect the other buttons of the same group.
+     *
+     * @private
+     * @memberof RadioButtonWidgetAnnotationElement
+     */
+    _toggle: function RadioButtonWidgetAnnotationElement_toggle(element) {
+      element.className = element.className &&
+        !this.data.noToggleToOff ?'' : 'checked';
+      var otherKids = this.layer.querySelectorAll(
+        '[data-parent-name="' + this.data.fullName.split('`')[0] + '"]');
+      // Unselect other kids
+      for(var i=0; i<otherKids.length; i++) {
```

Comment:
Change this to `for (var i = 0, ii = otherKids.length; i < ii; i++) {` to cache the length.